### ISSUE TITLE
database: Better understand the window of loss (and loss scenarios) for database backups

### ISSUE THREAD

**Issue by carmstrong** [deis/deis#2819]
The database component uses [wal-e](https://github.com/wal-e/wal-e) to ship backups and WAL logs to the deis-store component. Full backups occur by default every 3 hours, and WAL logs are shipped continuously. When a database comes up fresh on a host, it first asks WAL-e if there are any backups to restore. If so, it restores the backup and then begins to replay the WAL logs that were written since the backup.

There is potentially a small window of loss here if the database component should exit before the last WAL log is written to disk. Additionally, users (including @lorieri) have occasionally reported loss of some data when upgrading Deis or otherwise performing a database restore. We test Deis platform upgrades for every Deis release, and this includes a database restore. However, the amount of data we're storing is not large. There is likely either a larger loss window than we think, or other causes of data loss. 

Ceph doesn't seem to be the blame, as registry data (which is also stored in Ceph) is reported to be fully intact. This points to a combination of the database backup loop logic and wal-e itself. We should so some investigation to better attempt to replicate what users are sometimes seeing in real-world environments.

Somewhat related is the fragility of the backup loop in the bin/boot script: #2741

**Comment by lorieri** [comment]
Today I restarted the cluster, I started ceph in order (monitor, daemon, metadata, volume, gateway)

same happened, last app created did not appears in ```deis apps```

**Comment by carmstrong** [comment]
This should be re-evaluated after #2741 -- that reimplementation of the /bin/boot scripts is something we want to do as soon as possible, and that work will affect the findings from this.

**Comment by lorieri** [comment]
Hi,

I was thinking in something

lets say I start my database in machine A, it will create a database volume on A.
then A goes down, database will be migrate to B and gets backups from the gateway.
then all machines goes down and database starts on A before the gateway.

will it get the old volume and ignore backups ?

**Comment by bacongobbler** [comment]
yes, that's what is described in #3128.

**Comment by lorieri** [comment]
:)

**Comment by mboersma** [comment]
The fixes mentioned have been merged into Deis, and several other boundary cases relating to DB restore were found and fixed by the v1.10.1 release. I'm going to close this, but please open a new issue with specifics if something like this recurs with the current [Deis v1.11 release](https://github.com/deis/deis/releases/tag/v1.11.0).
